backend/portal/models.py

```diff
@@ -48,17 +50,52 @@ class Content(models.Model):
     admin_comment = models.CharField(max_length=255, null=True, blank=True)
     target_populations = models.ManyToManyField(TargetPopulation, blank=True)
     priority = models.IntegerField(default=0)
+    creator = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=True)
 
     class Meta:
         abstract = True
 
+    def _get_email_subject(self):
+        return f"[Portal] {self.__class__._meta.model_name.capitalize()} #{self.id}"
+
+    def _on_create(self):
+        send_automated_email.delay_on_commit(
+            self._get_email_subject(),
+            get_backend_manager_emails(),
+            (
+                f"A new {self.__class__._meta.model_name} for {self.club_code}"
+                f"has been created by {self.creator}"
+            ),
+        )
+
+    def _on_status_change(self):
+        send_automated_email.delay_on_commit(
+            self._get_email_subject(),
+            getattr(self.creator, "email", None),
+            f"Your {self.__class__._meta.model_name} status for {self.club_code} has been"
+            + "changed to {self.status}."
+            + (
+                f"\n\nAdmin comment: {self.admin_comment}"
+                if self.admin_comment and self.status == self.STATUS_REVISION
+                else ""
+            ),
+        )
+
+    def save(self, *args, **kwargs):
+        prev = self.__class__.objects.filter(id=self.id).first()
+        super().save(*args, **kwargs)
+        if prev is None:
+            return self._on_create()
```

Comment:
wait yes need for return here because i want need to hard stop since prev is None. 
it was originally 
`self._on_create()
return`

but i changed it to be fancy

Reply:
Wait then why don't you just have an `elif` for the latter clause?

Reply:
actually yea i could. I did return because i thought i would have more conditions to send an email

Reply:
yep up to you